sunkit_instruments/goes_xrs/goes_chianti_tem.py

```diff
@@ -0,0 +1,329 @@
+    temp = interpolate.splev(fluxratio, spline, der=0)
+
+    modelflux = (
+        response_table["FLONG_COR"][sat]
+        if abundance == "coronal"
+        else response_table["FLONG_PHO"][sat]
+    )
+
+    modeltemp = response_table["TEMP_MK"][sat]
+
+    spline = interpolate.splrep(modeltemp, modelflux * table_to_response_em, s=0)
+    denom = interpolate.splev(temp, spline, der=0)
+
+    emission_measure = longflux_corrected.value / denom
+
+    goes_times = goes_ts._data.index
+    df = pd.DataFrame(
+        {"temperature": temp, "emission_measure": emission_measure * 1e49},
+        index=goes_times,
+    )
+
+    units = {"temperature": u.MK, "emission_measure": u.cm ** (-3)}
+
+    header = {"Info": "Estimated temperature and emission measure"}
+
+    # return a new timeseries with temperature and emission measure
+    temp_em = ts.TimeSeries(df, header, units)
+
+    return temp_em
+
+
+def _manage_goesr_detectors(goes_ts, satellite_number, abundance="coronal"):
+    """
+    This manages which response to use for the GOES primary and secondary detectors used in the
+    observations for the GOES-R satellites (i.e. GOES 16 and 17).
+
+    The GOES 16- and 17- have two detectors for each channel to extend the dynamic range of observations,
+    namely XRS-A1, XRS-A2, XRS-B1, XRS-B2, for xrsa and xrsb, respectively.
+    During large flares, both the primary and secondary detectors may be used and given that each have a different
+    response, we need to match the data at individual times with the correct response.
+
+    Note that the primary channel conditions are values of 0,1,2,3 to indicate A1+B1, A2+B1, A1+B2, A2+B2 detector combos for GOES-R.
+    Here, we use the `xrsa{b}_primary_chan` columns to figure out which detectors are used for each timestep.
+
+    """
+
+    # These are the conditions for which detector combinations to use.
+    secondary_det_conditions = {0: [1, 1], 1: [2, 1], 2: [1, 2], 3: [2, 2]}
+
+    # here we split the timeseries into sections for which primary or secondary detectors are used
+    # and then calculate the response for each and then concatenate them back together.
+    outputs = []
+    for k in secondary_det_conditions:
+        dets = secondary_det_conditions[k]
+        (second_ind,) = np.where(
+            (goes_ts.quantity("xrsa_primary_chan") == dets[0])
+            & (goes_ts.quantity("xrsb_primary_chan") == dets[1])
+        )
+
+        goes_split = ts.TimeSeries(goes_ts._data.iloc[second_ind], goes_ts.units)
+
+        if len(goes_split._data) > 0:
+            output = _chianti_temp_emiss(
+                goes_split, satellite_number, abundance=abundance, secondary=int(k)
+            )
+            outputs.append(output)
+
+    if len(outputs) > 1:
+        full_output = outputs[0].concatenate(outputs[1:])
+    else:
+        full_output = outputs[0]
+
+    return full_output
+
+
+def calculate_radiative_loss(*args, **kwargs):
```

Comment:
This and the below NotImplemented function should be in the old module so people can find them where they expect to find them.